### PR TITLE
Swipe model

### PR DESCRIPTION
**Pull request by Jonathan-Bonnin**

### CODE REVIEW

File: GiTinder/Models/Swipe.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+namespace GiTinder.Models
+{
+    public class Swipe
+    {
+        [Key]
+        [JsonIgnore]
+        public int SwipeId { get; set; }
+        [Required]
+        [JsonProperty("user_id")]
+        public string UserId { get; set; }
```
Comment (adamgyulavari) [inline GiTinder/Models/Swipe.cs]: @Jonathan-Bonnin `User SwipingUser`

File: GiTinder/Models/Swipe.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        [Required]
+        [JsonProperty("swiped_user_id")]
+        public string SwipedUserId { get; set; }
```
Comment (adamgyulavari) [inline GiTinder/Models/Swipe.cs:17]: @Jonathan-Bonnin `User SwipedUser`

File: GiTinder/Models/Swipe.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        [Required]
+        [JsonProperty("direction")]
+        public SwipeDirection Direction { get; set; }
+        [Required]
+        [JsonProperty("timestamp")]
+        public string Timestamp { get; set; }
```
Comment (adamgyulavari) [inline GiTinder/Models/Swipe.cs]: @Jonathan-Bonnin this should be either a `Long` or `Timestamp`

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

**Comment by Jonathan-Bonnin** [comment]
@aze now the Id of Swipe is created by the combination of the 2 users:
```
modelBuilder.Entity<Swipe>() .HasKey(t => new { t.SwipingUserId, t.SwipedUserId });
```
in GiTinderContext.cs

Model + Tests adjusted according to our review + discussion

File: GiTinder/Models/Swipe.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+
+namespace GiTinder.Models
+{
+    public class Swipe
+    {
+        [Required]
+        [JsonProperty("swiping_user_id")]
```
Comment (adamgyulavari) [inline GiTinder/Models/Swipe.cs]: @Jonathan-Bonnin these are not going to be json properties

**Review by adamgyulavari — APPROVED** [review]

**Pull request merged**